Wangscape/logging/Logger.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "Level.h"
+
+#include <string>
+
+
+namespace logging
+{
+
+class Logger
+{
+public:
+    Logger() = default;
+    explicit Logger(std::ostream* out_)
+        : mAppender{out_}
+    {
+    }
+    void setStream(std::ostream* out)
+    {
+        mAppender = out;
+    }
+    template<typename T>
+    Logger& operator<<(T&& msg)
+    {
+        if (mAppender != nullptr) {
+            *mAppender << std::forward<T>(msg);
+        }
+        return *this;
+    }
+private:
+    std::ostream* mAppender = nullptr;
```

Comment:
It's just a non-owning pointer to an external (global) `std::ofstream`. It's not elegant, I know, but quite convenient. It saves from having additional classes hierarchy just for appenders and allows us to have simple generic `operator<<` implementation.